File: problems/2/python2/even_fibonacci_numbers.py
def F():
    a,b = 0,1
    while True:
        yield b
        a, b = b, a + b

def SubFib(startNumber, endNumber):
    for cur in F():
        if cur > endNumber: return
        if cur >= startNumber:
            yield cur

total = 0
for number in SubFib(1, 4000000):
    if number % 2 == 0:
        total += number
print "The sum of even numbers in fibonacci sequence bellow 4kk is {}".format(total)

File: problems/2/java1.8/src/main/java/com/gitbug/rogeralmeida/eulersjourney/problem2/EvenFibonacciNumbers.java
package com.gitbug.rogeralmeida.eulersjourney.problem2;

public class EvenFibonacciNumbers {


    public static void main(String[] args) {
        Integer number = 1;
        Integer previous = 0;
        Integer sum = 0;
        while (number < 4000000){
            if (number % 2 == 0){
                sum += number;
            }
            int current = number;
            number = previous + current;
            previous = current;
        }
        System.out.printf("The sum of even fibonacci numbers bellow 4kk is %d", sum);
    }
}
